### ISSUE TITLE
[Java] Gandiva Tests are failure related to Gandiva 

### ISSUE THREAD

**Issue by vibhatha** [apache/arrow#43576]
### Describe the bug, including details regarding any error messages, version, and platform.

At the moment, java-jars CI in Arrow Java is failing due to a Gandiva issue. This CI in green is much needed to determine the accuracy of certain PRs to Java. In order to keep it green, as [discussed](https://github.com/apache/arrow/pull/43503#issuecomment-2264988623) we are disabling the failing Gandiva tests temporarily. But needs to be re-enabled once the LLVM related issue is resolved. 

### Component(s)

Java

**Comment by jinchengchenghh** [comment]
https://github.com/apache/arrow/pull/43639#issuecomment-2285526638 This test also needs to be disabled.

**Comment by lidavidm** [comment]
@praveenbingo @pravindra you seem to be the last committers to have worked on the Java part of Gandiva; do you have any idea about this issue?

**Comment by vibhatha** [comment]
Most of the failing tests are due to the `Could not create LLJIT instance` issue [1]. Also, the ProjectTest is mainly failing once. So it would be better to disable the overall test suite and re-enable it once this has been properly fixed. 

[1] https://github.com/apache/arrow/actions/runs/10731577229/job/29761887531?pr=43978#step:7:20331

**Comment by lidavidm** [comment]
@laurentgo do you have any colleagues who might know what's up with Gandiva?

**Comment by raulcd** [comment]
I've moved this to 19.0.0, let me know if this should be a blocker for the release.

**Comment by lidavidm** [comment]
It's possible we're going to ship Gandiva in a known-broken state, so we should just highlight it in the release notes

**Comment by vibhatha** [comment]
+1 for this. I think that makes sense and a sub optimal solution for the problem we have.

**Comment by lriggs** [comment]
Hi, I work with @laurentgo at Dremio and I've started looking into this issue. Let me know if there is a better format for this discussion (email vs GH issue, etc). I wanted to summarize some of the different threads and then talk about how I've worked around the issue here at Dremio.

The test errors seem to all be "Symbols not found: [ llvm_orc_registerEHFrameSectionWrapper ]" which is also tracked [here](https://github.com/apache/arrow/issues/43769 ), and a few causes and ideas for fixes are listed [here](https://github.com/apache/arrow/pull/43770#issuecomment-2299793645) and [here](https://github.com/apache/arrow/issues/39695#issuecomment-1912186637) for Python.

I also found [this](https://github.com/llvm/llvm-project/issues/74671) issue regarding the symbols in the llvm project.

The poster there believes that requiring the symbols is a bad design. I tend to agree but I'm not sure why it was done in the first place.

The ORC symbol issue seems to stem from [https://github.com/apache/arrow/issues/37848](https://github.com/apache/arrow/issues/37848). We haven't had this problem at Dremio because we actually reverted that change in our local repo after I found a few llvm expression related test failures. I asked about that here [https://lists.apache.org/thread/xzrwx1t37dc88vo5yo5337l16ypqwcrd](https://lists.apache.org/thread/xzrwx1t37dc88vo5yo5337l16ypqwcrd). I spent considerable time looking into this issue but was never able to track it down.

I'm not sure what to do at this point. One idea is reverting the JIT upgrade while also following up with the llvm community on the symbol issue. @niyue Mentioned following up with them here, not sure if he found anything: [https://github.com/apache/arrow/issues/39695#issuecomment-1918780143](https://github.com/apache/arrow/issues/39695#issuecomment-1918780143)

**Comment by vibhatha** [comment]
Right. I think we can discuss the technical aspect of the issue here.
Also I started a ML thread to get community support to resolve this issue as this is a bit out of scope for me. 

FWIW, I would say it would be fine to get this to a release viable point by using an older version of an upgrade we have done so far. But I solely wouldn't be a judge for that. 

Really appreciate your support to look into this matter.

**Comment by niyue** [comment]
@lriggs Regarding the LLVM 17 symbol issue, I recall investigating it previously. It stems from a design change introduced in LLVM 17, and unfortunately, there doesn't appear to be an easy fix at that time. A related issue has been logged here: [LLVM Project Issue #74671](https://github.com/llvm/llvm-project/issues/74671).
